data/blog/tailscale-sucks.mdx

```diff
@@ -0,0 +1,464 @@
+network without a countersignature from an existing device, using a PKI that is
+separate from the Tailscale control plane.
+
+Perhaps you don't like that Tailscale only has OAuth sign-in, which requires you
+to use a Gmail, Microsoft, or GitHub account for auth (we support other
+providers too, but they're paid services so even less likely to be interesting
+to enthusiasts). That's fair, and we built [custom OIDC support][ts-oidc] so
+that you can bring your own auth stack if you prefer.
+
+And if you really cannot stand having a cloud involved in your vacuum robot at
+all, I understand. Given the track record of the SaaS/cloud industry, it's hard
+to trust anyone! In that case, [Headscale][headscale] is an open-source
+implementation of the Tailscale control plane that you can self-host, and cut
+Tailscale-the-company out of the loop entirely, while still getting the benefits
+of secure connectivity and NAT-busting remote access.
+
+## Conclusion
+
+So, what did we learn here? Well, I learned that jailbreaking vacuum cleaners
+feels great. I learned that Dennis Giese and Sören Beye are amazing people, and
+I'm grateful that they choose to share their knowledge and expertise with the
+world. I learned a whole lot about the state of IoT (remember the "S" in "IoT"
+stands for "security"), and the fascinating world of consumer robotics.
+
+And perhaps most importantly, I learned that if I blog about putting Tailscale
+on weird unusual devices, I can get Tailscale to reimburse me for the device.
+And that's a dangerous precedent to set.
+
+<figure className="float-none text-center sm:pl-4">
+  <Image
+    src="/images/tailscale-sucks/final.jpg"
+    width="621"
+    height="588"
+    alt="Photo of the Z10 robot in its charging dock. A Tailscale sticker has been applied to the top cover."
+  />
+  <figcaption>
+    Naturally, the cats have already figured out that the one place the robot can't clean up their
+    hair, is on top of the robot itself. The irony.
+  </figcaption>
+</figure>
+
+[mi-home]: https://en.wikipedia.org/wiki/Xiaomi_Smart_Home
+[giese-talk]: https://www.youtube.com/watch?v=DHsqb2poGII
+[giese-site]: https://dontvacuum.me/
+[beye-site]: https://hypfer.de/
+[valetudo]: https://valetudo.cloud/
+[hass]: https://www.home-assistant.io/
+[ifixit]: https://canada.ifixit.com/products/pro-tech-toolkit
+[valetudo-install]: https://valetudo.cloud/pages/installation/dreame.html
+[bus-pirate]: http://dangerousprototypes.com/docs/Bus_Pirate
+[ts-user]: https://tailscale.com/kb/1112/userspace-networking/
+[pkgs]: https://pkgs.tailscale.com/stable/#static
+[ts-small]: https://tailscale.com/kb/1207/small-tailscale/
+[ts-source]: https://github.com/tailscale/tailscale
+[scp]: https://lwn.net/Articles/835962/
+[ts-tag]: https://tailscale.com/kb/1068/acl-tags/
+[valetudo-patch]: https://github.com/danderson/Valetudo/commit/773b0d040dde113e5e9db390b30b09237fcbd89a
+[le]: https://letsencrypt.org/
+[ts-security]: https://tailscale.com/security
```

Comment:
might as well add the trailing slash and save people the redirect

Reply:
Done